### PR TITLE
fix(auth): accept x-api-key header

### LINKED ISSUE #36: Add x-api-key header support to auth middleware
## User Request

Claude Code agents send `x-api-key` header (Anthropic SDK standard) to the llm-proxy, but the auth middleware only accepts Ziti identity or `Authorization: Bearer` tokens, causing 401 errors.

### Root Cause
`internal/auth/middleware.go` → `resolveIdentity()` only checks:
1. Ziti identity from connection context
2. `Authorization: Bearer <token>` header

Claude Code (Anthropic SDK) sends `x-api-key: <token>` instead of `Authorization: Bearer <token>`. The middleware doesn't read this header, so it returns `"authorization required"` (401).

### Fix

**File: `internal/auth/middleware.go`**

1. Pass `r.Header.Get("x-api-key")` into `resolveIdentity` alongside `r.Header.Get("Authorization")`.

2. Update `resolveIdentity` signature:
```go
func resolveIdentity(
    ctx context.Context,
    authHeader string,
    apiKeyHeader string,   // NEW
    zitiResolver IdentityResolver,
    apiTokenResolver BearerTokenResolver,
) (context.Context, error)
```

3. After the Ziti identity check, try `Authorization: Bearer` first. If not present, fall back to `x-api-key` header value:
```go
accessToken, ok := httpauth.ExtractBearerToken(authHeader)
if !ok {
    apiKey := strings.TrimSpace(apiKeyHeader)
    if apiKey == "" {
        return ctx, errors.New("authorization required")
    }
    accessToken = apiKey
}
```

4. Update `Middleware` function call site:
```go
ctx, err := resolveIdentity(
    r.Context(),
    r.Header.Get("Authorization"),
    r.Header.Get("x-api-key"),    // NEW
    zitiResolver,
    apiTokenResolver,
)
```

### Verification
- `go build ./...`
- `go vet ./...`
- `go test ./...`

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- accept x-api-key as an auth header input
- fall back to x-api-key when bearer token is absent

## Testing
- buf generate buf.build/agynio/api --path agynio/api/llm/v1 --path agynio/api/users/v1 --path agynio/api/authorization/v1 --path agynio/api/ziti_management/v1 --path agynio/api/identity/v1
- go vet ./...
- go build ./...
- go test ./...

Fixes #36

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- buf generate buf.build/agynio/api --path agynio/api/llm/v1 --path agynio/api/users/v1 --path agynio/api/authorization/v1 --path agynio/api/ziti_management/v1 --path agynio/api/identity/v1
- go vet ./... (no issues)
- go build ./... (no issues)
- go test ./... (passed: 3 packages, failed: 0, skipped: 0)

File: internal/auth/middleware.go [hunk 1]
```diff
@@ -51,6 +58,10 @@ func resolveIdentity(ctx context.Context, authHeader string, zitiResolver Identi
 	}
 
 	accessToken, ok := httpauth.ExtractBearerToken(authHeader)
+	if !ok {
+		accessToken = strings.TrimSpace(apiKeyHeader)
+		ok = accessToken != ""
+	}
```
Comment (noa-lucent) [inline internal/auth/middleware.go:64]: [major] The `x-api-key` fallback logic is correct, but there are no unit tests for `resolveIdentity` at all — the `internal/auth/` package has zero `*_test.go` files, and the e2e tests only exercise `Authorization: Bearer`.

This is a security-sensitive authentication path. Please add unit tests for `resolveIdentity` covering at minimum:
1. `x-api-key` accepted when `Authorization` header is absent.
2. `Authorization: Bearer` takes precedence when both headers are present.
3. Empty/whitespace-only `x-api-key` still returns `"authorization required"`.
4. `x-api-key` value without the `agyn_` prefix is rejected with `"unsupported bearer token"`.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The implementation is clean, correct, and well-scoped — the fallback logic, trimming, and priority order are all sound. The token flows through the same `HasPrefix` + `ResolveFromToken` validation pipeline, so there's no security gap in the logic itself.

However, this is a security-sensitive authentication change with no unit test coverage. The `internal/auth/` package has no test files, and the e2e suite only exercises `Authorization: Bearer`. Please add unit tests for the new `x-api-key` path before merging.

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- buf generate buf.build/agynio/api --path agynio/api/llm/v1 --path agynio/api/users/v1 --path agynio/api/authorization/v1 --path agynio/api/ziti_management/v1 --path agynio/api/identity/v1
- go vet ./... (no issues)
- go build ./... (no issues)
- go test ./... (passed: 4 packages, failed: 0, skipped: 0)

**Review by noa-lucent — APPROVED** [review]
Tests cover all four requested scenarios: `x-api-key` accepted when `Authorization` is absent, `Bearer` takes precedence when both are present, blank API key rejected, and non-prefixed key rejected. The stub is clean and assertions are thorough.

Implementation and tests are both solid. Good to merge.
